score-data-format/src/main/java/io/cloudslang/content/json/services/JsonService.java

```diff
@@ -0,0 +1,154 @@
+        return generateResultingJsonString(wrappingQuote, jsonMap);
+    }
+
+    private String generateResultingJsonString(char wrappingQuote, Map<String, Object> jsonMap) {
+        JSONObject jsonObject = new JSONObject();
+
+        for(String key:jsonMap.keySet()){
+             jsonObject.put(key,jsonMap.get(key));
+        }
+
+        String newJson = jsonObject.toJSONString(JSONStyle.LT_COMPRESS);
+
+        if (newJson.charAt(1) != wrappingQuote) {
+            return replaceUnescapedOccurrencesOfCharacterInText(newJson, newJson.charAt(1), wrappingQuote);
+        }
+
+        return newJson;
+    }
+
+
+    private void parseJsonForInconsistencies(String normalizedJson) {
+        JsonProvider provider = new GsonJsonProvider();
+        Configuration configuration =  Configuration.builder().jsonProvider(provider).build();
+        JsonPath.parse(normalizedJson, configuration);       //throws an exception at runtime if the json is malformed
+    }
+
+    /**
+     * Returns the quote character used for specifying json member names and String values of json members
+     * @param jsonString the source json from which to extract the wrapping quote
+     * @return either one of the characters ' (single quote)or " (double quote)
+     */
+    private char retrieveWrappingQuoteTypeOfJsonMemberNames(String jsonString) {
+        char quote = '\"';   //  the default quote character used to specify json member names and string value according to the json specification
+        for (char c : jsonString.toCharArray()) {
+            if (c == '\'' || c == '\"') {
+                quote = c;
+                break;
+            }
+        }
+        return quote;
+    }
+
+    private void removeEmptyElementsFromMap(Map<String, Object> json) {
+        LinkedHashMap<String, Object> copyJson = new LinkedHashMap<String, Object>(json);
+        for (Object element : copyJson.values()) {
+            if (element == null) {
+                json.values().remove(element);
+            } else if (element instanceof String) {
+                if (element.equals("")) {
+                    json.values().remove(element);
+                }
+            } else if (element instanceof JSONArray) {
+                if (((JSONArray) element).isEmpty()) {
+                    json.values().remove(element);
+                } else {
+                    removeEmptyElementFromJsonArray((JSONArray) element);
+                }
+            } else if (element instanceof LinkedHashMap) {
+                if (((LinkedHashMap) element).isEmpty()) {
+                    json.values().remove(element);
+                } else {
+                    removeEmptyElementsFromMap((Map<String,Object>) element);
+                }
+            }
+        }
+    }
+
+    private void removeEmptyElementFromJsonArray(JSONArray jsonArray) {
+        JSONArray copyJsonArray = (JSONArray) jsonArray.clone();
+        for (Object element : copyJsonArray) {
```

Comment:
maybe you could use an iterator over the JSONArray here to iterate and remove elements without needing a clone of the array